.github/workflows/ci.yml

```diff
@@ -0,0 +1,148 @@
+name: CI
+
+on: [push]
+
+env:
+  IMAGE: ghcr.io/avtikhon/tarantool/testing/debian-stretch:latest
+  TRAVIS_MAKE: make -f .travis.mk
+
+jobs:
+  release:
```

Comment:
I'd suggest to split single workflow to several, for easier restart if anything goes wrong, i.e.:
- release, debug, release_lto;
- release_clang, release_lto_clang11, release_asan_clang11
- osx_10_15, osx_11_0
- and separate coverage run

Reply:
Not possible due to there are 2 stages, in your way luacheck from 1st stage will be duplicated.

Reply:
As I've answered elsewhere - luacheck is more suitable for separate flow. i.e. at the moment we would have 3 yamls:
- quick luacheck
- multiple linux builds (gcc/clang, lto/asan, etc);
- mac osx build (this should be separate because it's heavy and slow)

It's not obvious yet what should be done for proper coverall integration, because currently (as done in gcc/debug) it's not working yet, not attaching current reports to PR. Might require special flow also

Reply:
It is not clear why we should split current workflow. I have discussed it with Timur and I'll try to explain here.

The first problem is that GH Actions currently lack of feature with job restart. One can restart only a whole workflow, see documentation [1]. We often use job restart on Gitlab CI because something goes wrong (flaky tests, infrastructure problems etc) and we want to restart a single job without restarting a whole pipeline.

The second problem is that jobs on Mac are slower than on Linux because:
1. Macs provided by GH requires setup before building and testing.
2. Macs are less performant than Linux.

To mitigate both problems Timur suggest to split workflow suggested in a patch to at least two workflows:
- first one with jobs that will run on Linux
- second one with jobs that will run on Mac

@avtikhon, solution suggested by Timur looks like a compromise. Let's go with it if you have no another solution.

1. https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs#viewing-your-workflow-history

Reply:
> It is not clear why we should split current workflow. I have discussed it with Timur and I'll try to explain here.
> 
> The first problem is that GH Actions currently lack of feature with job restart. One can restart only a whole workflow, see documentation [1]. We often use job restart on Gitlab CI because something goes wrong (flaky tests, infrastructure problems etc) and we want to restart a single job without restarting a whole pipeline.
> 

That's not a problem, if we have flaky failing that it is really new fails - all flaky issues should be resolved in Q4'20.

> The second problem is that jobs on Mac are slower than on Linux because:
> 
> 1. Macs provided by GH requires setup before building and testing.
> 2. Macs are less performant than Linux.
> 

That's not true at all, please check timings at pipeline results:
https://github.com/tarantool/tarantool/actions/runs/420554916

while OSX runs in 13 minutes clang11 jobs test in 15 and 20 minutes.

> To mitigate both problems Timur suggest to split workflow suggested in a patch to at least two workflows:
> 
> * first one with jobs that will run on Linux
> * second one with jobs that will run on Mac
> 

Due to the problems above not true then we don't need it.

> @avtikhon, solution suggested by Timur looks like a compromise. Let's go with it if you have no another solution.
> 
> 1. https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs#viewing-your-workflow-history

What do you mean another solution ? for what ?

Reply:
This discussion was started because of the thesis that OSX jobs based on MAC hosts run longer than Linux jobs, the thesis was wrong - some Linux jobs are longer than OSX. Please resolve this discussion due to I don't see any more objections to divide pipeline on standalone jobs, or please provide anything that I didn't answer.

Reply:
Now you understand why I've suggested to split gcc and clang jobs originally?

>  - release, debug, release_lto;
>  - release_clang, release_lto_clang11, release_asan_clang11
>  - osx_10_15, osx_11_0
> -  and separate coverage run

Reply:
Jobs divided.